source/type/message.cpp

```diff
@@ -523,7 +517,7 @@ const Disruption* DisruptionHolder::get_disruption(const std::string& uri) const
     return it->second.get();
 }
 
-void DisruptionHolder::add_weak_impact(boost::weak_ptr<Impact> weak_impact) {
+void DisruptionHolder::add_weak_impact(const boost::weak_ptr<Impact>& weak_impact) {
```

Comment:
Not really sure if a const ref to a weak_ptr and a copy of a weak_ptr does the same thing ?